reporter/v2/pkg/reporter/task.go

```diff
@@ -329,6 +334,26 @@ func getMeterDefinitionReferences(
 	return
 }
 
+// Stop() and Sleep to allow queue to write out events.
+// Calling Broadcaster Shutdown() otherwise is a risk of panic and lost event
+// Until we are using a newer k8s api-machinery version
+// https://github.com/kubernetes/kubernetes/issues/94906
+func provideReporterEventBroadcaster(kubeclientset kubernetes.Interface) (record.EventBroadcaster, func()) {
+	eventBroadcaster := record.NewBroadcaster()
+	eventBroadcaster.StartStructuredLogging(0)
+	shutdownInterface := eventBroadcaster.StartRecordingToSink(&typedcorev1.EventSinkImpl{Interface: kubeclientset.CoreV1().Events("")})
+	return eventBroadcaster, func() {
+		shutdownInterface.Stop()
+		logger.Info("Wait for event broadcaster to stop...")
+		time.Sleep(10 * time.Second)
+	}
```

Comment:
Nice use of a callback. There is a kubernetes Closer concept but it's hard to implement, function works fine.